bfh/fields.py

```diff
@@ -56,11 +60,19 @@ def __init__(self, required=True):
             required (Bool) - is this a required field in the schema?
         """
         self.required = required
+        self.default = default
 
     def __get__(self, instance, cls=None):
         if instance is None:
             return self
-        return instance.__dict__.get(self.field_name)
+
+        value = instance.__dict__.get(self.field_name)
+        if value is not None:
+            return value
+        else:
+            default = get_default(self.default)
+            instance.__dict__[self.field_name] = default
+            return default
```

Comment:
@benauthor I moved the logic to the `__get__`, although it's a bit weird that we are actually setting the default to the value right here.. we have to if we don't want to do it in the `__set__`. The reason being is that we need to keep track of the specific instance of default that we are currently using. If we just respond with a default and don't assign it... we will have no memory of it, and can't edit it.